pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/DownloadParticipantListRoute.java

```diff
@@ -0,0 +1,181 @@
+
+import com.google.gson.Gson;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.util.CellRangeAddress;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.broadinstitute.dsm.model.ParticipantColumn;
+import org.broadinstitute.dsm.model.elastic.sort.Alias;
+import org.broadinstitute.dsm.model.excel.ParticipantRecord;
+import org.broadinstitute.dsm.model.filter.FilterFactory;
+import org.broadinstitute.dsm.model.filter.Filterable;
+import org.broadinstitute.dsm.model.participant.DownloadParticipantListPayload;
+import org.broadinstitute.dsm.model.participant.ParticipantWrapperDto;
+import org.broadinstitute.dsm.model.participant.ParticipantWrapperResult;
+import org.broadinstitute.dsm.security.RequestHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+public class DownloadParticipantListRoute extends RequestHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(DownloadParticipantListRoute.class);
+    private static final String FILE_DATE_FORMAT = "ddMMyyyy";
+
+    public DownloadParticipantListRoute() {
+    }
+
+    @Override
+    public Object processRequest(Request request, Response response, String userId) throws Exception {
+        DownloadParticipantListPayload payload = new Gson().fromJson(request.body(), DownloadParticipantListPayload.class);
+
+        List<ParticipantColumn> columnNames = payload.getColumnNames();
+        List<String> esAliases = columnNames.stream()
+                .map(column -> {
+                    Alias esAlias;
+                    if (column.getObject() != null) {
+                        esAlias = Alias.of(column.getObject());
+                    } else {
+                        esAlias = Alias.of(column.getTableAlias());
+                    }
+                    return esAlias.getValue().isEmpty()? column.getName() : esAlias.getValue() + "." + column.getName();
+                })
+                .collect(Collectors.toList());
+
+        Filterable filterable = FilterFactory.of(request);
+        ParticipantWrapperResult filteredList = (ParticipantWrapperResult) filterable.filter(request.queryMap());
+        Workbook workbook = new XSSFWorkbook();
+        List<String> headerNames = payload.getHeaderNames();
+        Sheet sheet = workbook.createSheet("Participant List");
+        Row header = sheet.createRow(0);
+        IntStream.range(0, headerNames.size()).forEach(i -> {
+            Cell cell = header.createCell(i);
+            cell.setCellValue(headerNames.get(i));
+            sheet.autoSizeColumn(i);
+        });
+        int currentRow = 1;
+        for (ParticipantWrapperDto participant : filteredList.getParticipants()) {
+            List<Object> excelRow = new ArrayList<>();
```

Comment:
is there any chance to introduce some abstraction instead of having `Object`? something like interface which would be implemented by concrete classes and would have one abstract method which would produce necessary data?